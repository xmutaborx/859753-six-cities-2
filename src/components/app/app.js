import React, {Fragment} from 'react';
import PropTypes from 'prop-types';

const App = ({titles, handleClick}) => {

  return (
    <div>
      {titles.map((title, i) =>
        <Fragment key={title + i}>
          <p onClick={handleClick}>{title}</p>
          <img src="img/amsterdam.jpg" style={{width: `100px`, height: `150px`}}/>
        </Fragment>
      )};
    </div>
  );
};

App.propTypes = {
  titles: PropTypes.array.isRequired,
  handleClick: PropTypes.func
};

App.defaultProps = {
  titles: [`Default title`]
};

export default App;
