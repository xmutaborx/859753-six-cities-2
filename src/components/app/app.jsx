import React, {PureComponent} from 'react';
import PropTypes from 'prop-types';
import {connect} from 'react-redux';
import {ActionCreator} from '../../reducer/reducer';

import OffersList from '../offer-list/offer-list.jsx';
import CitiesMap from '../cities-map/cities-map.jsx';
import CitiesList from '../cities-list/cities-list.jsx';

const MAX_CITY = 6;

class App extends PureComponent {
  _setAvailableCityFromOffers(offers) {
    const list = offers.map((offer) => offer.city.name);
    const availableCities = Array.from(new Set(list));

    this.props.changeCity(availableCities[0]);
    this.props.setAvailableCities(availableCities);
  }

  _setAvailableOffers(offers, city) {
    const availableOffers = offers.filter((offer) => offer.city.name === city);
    this.props.setAvailableOffers(availableOffers);
  }

  _listOfCities() {
    return this.props.availableCities.slice(0, MAX_CITY);
  }

  _listOfPins() {
    return this.props.availableOffers.map((offer) => offer.coordinates);
  }

  componentDidMount() {
    // this._setAvailableCityFromOffers(this.props.offers);
    // this._setAvailableOffers(this.props.offers, this.props.city);
  }

  componentDidUpdate(prevProps) {
    if (prevProps.offers.length !== this.props.offers.length) {
      this._setAvailableCityFromOffers(this.props.offers);

      this._setAvailableOffers(this.props.offers, this.props.city);
    }
  }

  render() {
    const {mapConfig, changeCity, city} = this.props;

    return (
      <div className="page page--gray page--main">
        <main className="page__main page__main--index">
          <CitiesList
            cities={this._listOfCities()}
            changeCity={changeCity}
          />
          <div className="cities">
            <div className="cities__places-container container">
              <OffersList cards={this.props.offers} city={city} />
              <div className="cities__right-section">
                <section className="cities__map map">
                  <CitiesMap mapConfig={mapConfig} pins={this._listOfPins()} />
                </section>
              </div>
            </div>
          </div>
        </main>
      </div>
    );
  }
}

App.propTypes = {
  offers: PropTypes.array,
  city: PropTypes.string,
  changeCity: PropTypes.func.isRequired,
  setAvailableCities: PropTypes.func.isRequired,
  availableCities: PropTypes.array.isRequired,
  setAvailableOffers: PropTypes.func.isRequired,
  availableOffers: PropTypes.arrayOf(PropTypes.object),
  mapConfig: PropTypes.shape({
    defaultCity: PropTypes.arrayOf(PropTypes.number).isRequired,
    zoom: PropTypes.number.isRequired,
    layer: PropTypes.string.isRequired,
    copyRight: PropTypes.string.isRequired
  }),
};


const mapStateToProps = (state, ownProps) => Object.assign({}, ownProps, {
  city: state.city,
  offers: state.offers,
  availableCities: state.availableCities,
  availableOffers: state.availableOffers,
});

const mapDispatchToProps = (dispatch) => ({
  // setOffers: (offers) => dispatch(ActionCreator.setOffers(offers)),
  changeCity: (city) => dispatch(ActionCreator.changeCity(city)),
  setAvailableCities: (cities) => dispatch(ActionCreator.setAvailableCities(cities)),
  setAvailableOffers: (offers) => dispatch(ActionCreator.setAvailableOffers(offers)),
});

export {App};
export default connect(mapStateToProps, mapDispatchToProps)(App);
