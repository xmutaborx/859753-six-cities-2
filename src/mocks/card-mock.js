export const cardMock = [
  {
    id: 42,
    title: `test title`,
    price: 420,
    type: `type`,
    premium: true,
    img: `img/apartment-01.jpg`,
    rating: 42,
    coordinates: [12, 42]
  }
];
