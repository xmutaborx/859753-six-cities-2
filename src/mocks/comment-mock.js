/* eslint-disable camelcase */
const COMMENT_MOCK = [
  {
    comment: `test comment`,
    date: `08.12.2019`,
    id: 69,
    rating: 5,
    user: {
      avatar_url: `src`,
      id: 42,
      is_pro: true,
      name: `Name`
    }
  }
];

export default COMMENT_MOCK;
