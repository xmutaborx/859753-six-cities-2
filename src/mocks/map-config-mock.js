export const mapConfigMock = {
  defaultCity: [12.12345, 1.2],
  defaultCord: [12.12345634556, 4.89309666406198],
  zoom: 12,
  icon: {
    iconUrl: `img/pin.svg`,
    iconSize: [12, 12]
  },
  zoomControl: false,
  marker: true,
  layer: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  copyRight: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`,
};
