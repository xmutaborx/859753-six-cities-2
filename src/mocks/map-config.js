export const mapConfig = {
  defaultCity: [52.38333, 4.9],
  zoom: 12,
  layer: `https://{s}.basemaps.cartocdn.com/rastertiles/voyager/{z}/{x}/{y}{r}.png`,
  copyRight: `&copy; <a href="https://www.openstreetmap.org/copyright">OpenStreetMap</a> contributors &copy; <a href="https://carto.com/attributions">CARTO</a>`,
};
