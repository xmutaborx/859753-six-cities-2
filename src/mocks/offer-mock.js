const OFFER_MOCK = [
  {
    id: 1,
    city: {
      name: `Amsterdam`,
      location: {
        latitude: 52.370216,
        longitude: 4.895168,
        zoom: 10
      }
    },
    // eslint-disable-next-line camelcase
    preview_image: `img/1.png`,
    images: [`img/1.png`, `img/2.png`],
    title: `Beautiful & luxurious studio at great location`,
    // eslint-disable-next-line camelcase
    is_favorite: false,
    // eslint-disable-next-line camelcase
    is_premium: false,
    rating: 4.8,
    type: `apartment`,
    bedrooms: 3,
    // eslint-disable-next-line camelcase
    max_adults: 4,
    price: 120,
    goods: [`Heating`, `Kitchen`, `Cable TV`, `Washing machine`, `Coffee machine`, `Dishwasher`],
    host: {
      id: 3,
      // eslint-disable-next-line camelcase
      is_pro: true,
      name: `Angelina`,
      // eslint-disable-next-line camelcase
      avatar_url: `img/1.png`
    },
    description: `A quiet cozy and picturesque that hides behind a a river by the unique lightness of Amsterdam.`,
    location: {
      latitude: 52.35514938496378,
      longitude: 4.673877537499948,
      zoom: 8
    }
  }
];

export default OFFER_MOCK;
