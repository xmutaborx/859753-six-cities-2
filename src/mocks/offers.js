export const offers = [
  {
    id: 1,
    title: `Beautiful & luxurious apartment at great location`,
    price: 120,
    type: `apartment`,
    premium: true,
    img: `img/apartment-01.jpg`,
    rating: 93,
    coordinates: [52.3909553943508, 4.85309666406198]
  },
  {
    id: 2,
    title: `Wood and stone place`,
    price: 80,
    type: `Private room`,
    premium: false,
    img: `img/room.jpg`,
    rating: 80,
    coordinates: [52.369553943508, 4.85309666406198]
  },
  {
    id: 3,
    title: `Canal View Prinsengracht`,
    price: 132,
    type: `apartment`,
    premium: false,
    img: `img/apartment-02.jpg`,
    rating: 80,
    coordinates: [52.3909553943508, 4.929309666406198]
  },
  {
    id: 4,
    title: `Nice, cozy, warm big bed apartment`,
    price: 180,
    type: `apartment`,
    premium: true,
    img: `img/apartment-03.jpg`,
    rating: 100,
    coordinates: [52.3809553943508, 4.939309666406198]
  }
];
