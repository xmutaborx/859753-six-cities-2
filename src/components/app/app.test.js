import React from 'react';
import renderer from 'react-test-renderer';
import {App} from './app';

import {cardMock} from '../../mocks/card-mock';
import {mapConfigMock} from '../../mocks/map-config-mock';

it(`App correctly renders`, () => {
  const div = global.document.createElement(`div`);
  div.id = `map`;
  global.document.body.appendChild(div);
  const tree = renderer
    .create(<App
      mapConfig={mapConfigMock}
      city={`city_name`}
      offers={cardMock}
      availableCities={[`first city`, `second city`]}
      availableOffers={cardMock}
      changeCity={() => {}}
      setOffers={() => {}}
      setAvailableCities={() => {}}
      setAvailableOffers={() => {}}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});

