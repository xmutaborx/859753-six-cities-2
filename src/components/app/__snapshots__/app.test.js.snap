// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App correctly renders 1`] = `
Array [
  <div
    className="tabs"
  >
    <section
      className="locations container"
    >
      <ul
        className="locations__list tabs__list"
      >
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Paris
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Cologne
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Brussels
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Amsterdam
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Hamburg
            </span>
          </a>
        </li>
        <li
          className="locations__item"
        >
          <a
            className="locations__item-link tabs__item"
            href="#"
            onClick={[Function]}
          >
            <span>
              Dusseldorf
            </span>
          </a>
        </li>
      </ul>
    </section>
  </div>,
  <div
    className="cities"
  >
    <div
      className="cities__places-container container"
    >
      <section
        className="cities__places places"
      >
        <h2
          className="visually-hidden"
        >
          Places
        </h2>
        <b
          className="places__found"
        >
          0
           places to stay in 
          city_name
        </b>
        <div
          className="cities__places-list places__list tabs__content"
        />
      </section>
      <div
        className="cities__right-section"
      >
        <div
          className="cities__map"
          id="map"
        />
      </div>
    </div>
  </div>,
]
`;
